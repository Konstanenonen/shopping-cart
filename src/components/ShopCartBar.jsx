import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/components/ShopCartBar.module.scss';

function ShopCartBar({ itemsAmount }) {
  return (
    <div className={styles.container}>
      <p>Items in shopping cart: {itemsAmount}</p>
      <button type="button">CHECKOUT</button>
    </div>
  );
}

ShopCartBar.propTypes = {
  itemsAmount: PropTypes.number.isRequired,
};

export default ShopCartBar;
