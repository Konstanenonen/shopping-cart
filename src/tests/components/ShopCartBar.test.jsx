/* global it describe expect */

import React from 'react';
import { render } from '@testing-library/react';
import ShopCartBar from '../../components/ShopCartBar';

describe('ShopCartBar component', () => {
  it('renders itemsAmount that is given as a prop', () => {
    const { container } = render(<ShopCartBar itemsAmount={42} />);
    expect(container).toMatchSnapshot();
  });
});
