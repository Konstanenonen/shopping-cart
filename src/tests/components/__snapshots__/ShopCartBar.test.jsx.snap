// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopCartBar component renders itemsAmount that is given as a prop 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      Items in shopping cart: 
      42
    </p>
    <button
      class="button"
      type="button"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
